source/game.roomhandler.scriptagency.bmx

```diff
@@ -727,14 +727,20 @@ Type RoomHandler_ScriptAgency extends TRoomHandler
 				'of the same base template (high similarity)
 				For Local i:Int = 0 Until 10
 					script = GetScriptCollection().GetRandomAvailable(usedTemplateIDs)
+
 					If preventNewGoodScript And script.GetPotential() > 0.3 Then
 						GetScriptCollection().Remove(script)
-					ElseIf script.getTitle().contains("#") and i < 10
+						script = Null
+					ElseIf script.getTitle().contains("#")
 						GetScriptCollection().Remove(script)
+						script = Null
 					Else
 						Exit
 					Endif
 				Next
+				'if the prevention was not successful, simply fall back
+				'to using a random without additional requirements
+				If Not script Then script = GetScriptCollection().GetRandomAvailable(usedTemplateIDs)
```

Comment:
Ich frage mich, ob es statt diesem Fallback nicht sinnvoller wäre, die Schleife selbst umzubauen. Pseudocode pro Slot:
```
script=null
while script = null
  script= createRandom
  if(shouldNotUseScript) then removeScript;script=null
endwhile
```
Die 10 Versuche sind wie an anderen Stellen auch vergleichsweise arbiträr. Mittlerweile gibt es auch so viele Vorlagen, dass es eine geben sollte, die die Kriterien erfüllen sollte.

Reply:
```
				'prevent cheat of buying scripts to get good ones
				'on normal refill, only offer pool scripts or none-prime ones
				Local preventNewGoodScript:Int = Not replaceOffer And (GetScriptCollection().GetAvailableScriptList().Count() = 0)
				'get a new script - but avoid having multiple scripts
				'of the same base template (high similarity)
				For Local i:Int = 0 Until 100
					script = GetScriptCollection().GetRandomAvailable(usedTemplateIDs)
					
					'unset script if criterias do not fit
					If script and preventNewGoodScript And script.GetPotential() > 0.3 Then
						GetScriptCollection().Remove(script)
						script = Null
					EndIf
					If script and script.getTitle().contains("#")
						GetScriptCollection().Remove(script)
						script = Null
					EndIf
					
					If script Then Exit
				Next
				'if the prevention was not successful, simply fall back
				'to using a random without additional requirements
				If Not script Then script = GetScriptCollection().GetRandomAvailable(usedTemplateIDs)				
```
Ob nun while/repeat-loop oder for mit grossem "to" ... das Prinzip ist das gleiche.

Reply:
Zugegeben, aber im Moment sind es nur 10 Versuche. Sollte das angehoben werden?